File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int result = -1;
        int needleSize = needle.size();
        int index = 0;
        for(int i=0;i<haystack.size();i++){
            if(haystack[i] == needle[index]){
                index++;
            }
            else {
                i = i - index;
                index = 0;
            }
            if(index == needleSize){
                result = i+1 - index;
                break;
            }
        }
        return result;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int needleSize = needle.length();
        for(int i=0;i<=haystack.length()-needleSize;i++){
            String temp = haystack.substring(i, i + needleSize);
            if(temp.equals(needle)) return i;
        }
        return -1;
    }
}